db-actions.ts

```diff
@@ -0,0 +1,100 @@
+import {Product} from './db-schema';
+import {logger} from './logger';
+
+const retrieveAllProducts = () => {
```

Comment:
all those function returns Promise . Promise is a generic type - add the generic type (`Promise<Product>`)

Reply:
I added a type for a productDocument and for the object which returns from toClient() method.
It resolved all ts problems